tests/prefetch_related/tests.py

```diff
@@ -695,6 +695,14 @@ def test_custom_qs(self):
         ).first()
         self.assertIsNone(room.main_room_of_attr)
 
+        person = Person.objects.prefetch_related(
+            Prefetch('houses', queryset=House.objects.filter(name='House 1')),
+        ).get(pk=self.person1.pk)
+        self.assertEqual(
+            list(person.houses.all()),
+            list(person.houses.all().all()),
+        )
```

Comment:
The fact this assertion fails without the fix shows how the POLA is not respected here.